src/main/java/com/sap/sgs/phosphor/fosstars/data/github/PackageManagement.java

```diff
@@ -0,0 +1,232 @@
+   * Maps a package manager to a list of its possible config files.
+   *
+   * @see <a href="https://help.github.com/en/github/visualizing-repository-data-with-graphs/listing-the-packages-that-a-repository-depends-on">
+   *      Listing the packages that a repository depends on</a>
+   */
+  private static final Map<PackageManager, Predicate<String>[]> CONFIG_FILES_PATTERNS
+      = new EnumMap<>(PackageManager.class);
+
+  static {
+    register(MAVEN, "pom.xml"::equals);
+    register(GRADLE, "build.gradle"::equals);
+    register(NPM, "package-lock.json"::equals, "package.json"::equals);
+    register(YARN, "yarn.lock"::equals, "package.json"::equals);
+    register(DOTNET, ".csproj"::equals, ".vbproj"::equals, ".nuspec"::equals,
+        ".vcxproj"::equals, ".fsproj"::equals, "packages.config"::equals);
+    register(RUBYGEMS, "Gemfile.lock"::equals, "Gemfile"::equals, ".gemspec"::endsWith);
+    register(COMPOSER, "composer.json"::equals, "composer.lock"::equals);
+  }
+
+  /**
+   * Adds a number of known package managers for a language.
+   *
+   * @param language The language.
+   * @param list The package managers.
+   */
+  private static void register(Language language, PackageManager... list) {
+    if (KNOWN_PACKAGE_MANAGERS.containsKey(language)) {
+      throw new IllegalArgumentException(String.format("%s is already there!", language));
+    }
+    KNOWN_PACKAGE_MANAGERS.put(language, new PackageManagers(list));
+  }
+
+  /**
+   * Adds a list of possible config files for a package manager.
+   *
+   * @param packageManager The package manager.
+   * @param matchers A list of matchers for the possible config files.
+   */
+  private static void register(PackageManager packageManager, Predicate<String>... matchers) {
+    if (CONFIG_FILES_PATTERNS.containsKey(packageManager)) {
+      throw new IllegalArgumentException(String.format("%s is already there!", packageManager));
+    }
+    CONFIG_FILES_PATTERNS.put(packageManager, matchers);
+  }
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param where A GitHub organization of user name.
+   * @param name A name of a repository.
+   * @param github An interface to the GitHub API.
+   */
+  public PackageManagement(String where, String name, GitHub github) {
+    super(where, name, github);
+  }
+
+  @Override
+  public PackageManagement update(ValueSet values) throws IOException {
+    Objects.requireNonNull(values, "Hey! Values can't be null!");
+    logger.info("Looking for package managers ...");
+    values.update(packageManagers());
+    return this;
+  }
+
+  /**
+   * Gets a list of package managers that are used in a project.
+   *
+   * @return A value of the feature
+   *         {@link com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures#PACKAGE_MANAGERS}.
+   * @throws IOException If something went wrong.
+   */
+  private Value<PackageManagers> packageManagers() throws IOException {
+    Optional<Value> something = cache().get(url, PACKAGE_MANAGERS);
```

Comment:
Where is url defined? I couldn't find it. or am I missing something?

Reply:
oh it's in AbstractGitHubDataProvider?

Reply:
The `url` field is in the `AbstractGitHubDataProvider` class. This field is going to be removed in #112